tests/integration/helpers.py

```diff
@@ -248,17 +293,22 @@ async def access_all_dashboards(
     stop=stop_after_attempt(5),
     wait=wait_fixed(15),
     retry_error_callback=lambda _: False,
-    retry=lambda x: x is False,
+    retry=retry_if_result(lambda x: x is False),
 )
 def get_dashboard_ca_cert(model_full_name: str, unit: str):
-    output = subprocess.run(
-        [
-            "bash",
-            "-c",
-            f"JUJU_MODEL={model_full_name} juju scp "
-            f"ubuntu@{unit}:/var/snap/opensearch-dashboards/current/etc/opensearch-dashboards/certificates/ca.pem ./",
-        ],
-    )
+    try:
+        output = subprocess.run(
+            [
+                "bash",
+                "-c",
+                f"JUJU_MODEL={model_full_name} juju scp "
```

Comment:
You can do the same with [libjuju scp_{to|from}](https://github.com/juju/python-libjuju/blob/6c6d70d483937cec4404116c376b926106e05835/juju/unit.py#L313-L351)

Reply:
I can give it a try (again. I recall I had an attempt to use it, and I came across missing options that can be directly passed to `scp` while not available when using the `libjuju` implementaiton of it)